### ISSUE TITLE
Crash with "GPU process (undefined) crashed (133)" in Wayland session

### ISSUE THREAD

**Issue by develop7** [flathub/com.plexamp.Plexamp#60]
The Plexamp crashes for me upon start

Repro:

1. `flatpak run com.plexamp.Plexamp`

Expected: plexamp window shows up

Instead: plexamp doesn't show up, producing following console output:

<details>
<summary>Console output</summary>

```
$  flatpak run com.plexamp.Plexamp 
WAYLAND enabled, using the Wayland Electron backend
Using NVIDIA on Wayland, applying workaround
Passing the following arguments to Electron: --enable-gpu-rasterization --enable-zero-copy --enable-gpu-compositing --enable-native-gpu-memory-buffers --enable-oop-rasterization -ozone-platform=wayland --disable-gpu-sandbox --enable-features=UseSkiaRenderer,UseOzonePlatform,WaylandWindowDecorations
LaunchProcess: failed to execvp:
xdg-settings
[3:0529/160548.579021:ERROR:bus.cc(398)] Failed to connect to the bus: Failed to connect to socket /run/dbus/system_bus_socket: No such file or directory
[3:0529/160548.579070:ERROR:bus.cc(398)] Failed to connect to the bus: Failed to connect to socket /run/dbus/system_bus_socket: No such file or directory
Main: restoring window state from [object Object]/window-state-plexamp-main.json
Started Media Service: false
[50:0529/160548.625198:ERROR:gpu_init.cc(481)] Passthrough is not supported, GL is egl, ANGLE is 
Linux: Registering MPRIS service.
[50:0529/160548.664887:ERROR:gbm_pixmap_wayland.cc(76)] Cannot create bo with format= RGBA_8888 and usage=SCANOUT
[50:0529/160548.664904:ERROR:gpu_memory_buffer_factory_native_pixmap.cc(198)] Failed to create pixmap 768x1024, RGBA_8888, usage SCANOUT
[50:0529/160548.664919:ERROR:gbm_pixmap_wayland.cc(76)] Cannot create bo with format= RGBA_8888 and usage=GPU_READ
[50:0529/160548.664924:ERROR:gpu_memory_buffer_factory_native_pixmap.cc(198)] Failed to create pixmap 768x1024, RGBA_8888, usage GPU_READ
[50:0529/160548.664928:ERROR:shared_image_backing_factory_gl_image.cc(359)] CreateSharedImage: Failed to create bindable image
[50:0529/160548.664932:ERROR:shared_image_factory.cc(697)] CreateSharedImage: could not create backing.
[50:0529/160548.664983:ERROR:shared_context_state.cc(538)] Failed to make current since context is marked as lost
[50:0529/160548.664989:ERROR:skia_output_surface_impl_on_gpu.cc(1744)] Failed to make current.
[50:0529/160548.665090:ERROR:raster_decoder.cc(1252)]   RasterDecoderImpl: Context lost during MakeCurrent.
[50:0529/160548.665254:ERROR:shared_context_state.cc(538)] Failed to make current since context is marked as lost
[50:0529/160548.665259:ERROR:skia_output_surface_impl_on_gpu.cc(1744)] Failed to make current.
[50:0529/160548.665380:ERROR:raster_decoder.cc(1252)]   RasterDecoderImpl: Context lost during MakeCurrent.
[50:0529/160548.665393:ERROR:raster_decoder.cc(1252)]   RasterDecoderImpl: Context lost during MakeCurrent.
[50:0529/160548.665398:ERROR:raster_decoder.cc(1252)]   RasterDecoderImpl: Context lost during MakeCurrent.
[50:0529/160548.668520:ERROR:gbm_pixmap_wayland.cc(76)] Cannot create bo with format= RGBA_8888 and usage=SCANOUT
[50:0529/160548.668529:ERROR:gpu_memory_buffer_factory_native_pixmap.cc(198)] Failed to create pixmap 768x1024, RGBA_8888, usage SCANOUT
[50:0529/160548.668535:ERROR:gbm_pixmap_wayland.cc(76)] Cannot create bo with format= RGBA_8888 and usage=GPU_READ
[50:0529/160548.668541:ERROR:gpu_memory_buffer_factory_native_pixmap.cc(198)] Failed to create pixmap 768x1024, RGBA_8888, usage GPU_READ
[50:0529/160548.668544:ERROR:shared_image_backing_factory_gl_image.cc(359)] CreateSharedImage: Failed to create bindable image
[50:0529/160548.668549:ERROR:shared_image_factory.cc(697)] CreateSharedImage: could not create backing.
[3:0529/160548.947494:ERROR:gpu_process_host.cc(974)] GPU process exited unexpectedly: exit_code=133
Error: GPU process (undefined) crashed (133) :(
Quitting...
```

</details>

<details>
<summary>System information</summary>

```
$ inxi -xx -S -G
System:
  Host: t-rex Kernel: 6.3.2-1-default arch: x86_64 bits: 64 compiler: gcc
    v: 13.0.1 Desktop: GNOME v: 44.1 tk: GTK v: 3.24.38 wm: gnome-shell dm: GDM
    Distro: openSUSE Tumbleweed 20230527
Graphics:
  Device-1: NVIDIA GA102 [GeForce RTX 3080 Lite Hash Rate]
    vendor: Micro-Star MSI driver: nvidia v: 525.116.04 arch: Ampere pcie:
    speed: 2.5 GT/s lanes: 16 ports: active: none off: DP-1,HDMI-A-1
    empty: DP-2,DP-3 bus-ID: 08:00.0 chip-ID: 10de:2216
  Display: wayland server: X.org v: 1.21.1.8 with: Xwayland v: 23.1.1
    compositor: gnome-shell driver: X: loaded: nvidia
    unloaded: fbdev,modesetting,vesa alternate: nouveau,nv gpu: nvidia
    d-rect: 3640x3360 display-ID: 0
  Monitor-1: DP-1 pos: bottom-l res: 2560x1440 size: N/A
  Monitor-2: HDMI-1 pos: top-right res: 1080x1920 size: N/A
  API: OpenGL v: 4.6.0 NVIDIA 525.116.04 renderer: NVIDIA GeForce RTX
    3080/PCIe/SSE2 direct-render: Yes
```

</details>

**Comment by user0753** [comment]
Getting this too with identical console output. Hard/Software is slightly different w/ a GTX 1080, NVIDIA 525.89.02, Kernel 6.1.31-lts, and Plasma 5.27.5 (also on Wayland). Can't really try the NVIDIA 530 or 535 series because they break Wayland entirely for me. Tried downgrading xwayland a couple versions but no luck.

Works fine in an X11 session though.

**Comment by develop7** [comment]
Okay, the `--disable-gpu-sandbox` breaks it for me

Dropping this key results in Plexamp working despite numerous complains by `MESA-LOADER` in console (see below)

<details>
<summary>Console output</summary>

```text
[📦 com.plexamp.Plexamp ~]$  /app/bin/zypak-wrapper /app/bin/plexamp/plexamp --enable-gpu-rasterization --enable-zero-copy --enable-gpu-compositing --enable-native-gpu-memory-buffers --enable-oop-rasterization --ozone-platform-hint=auto --enable-features=UseSkiaRenderer,UseOzonePlatform,WaylandWindowDecorations
LaunchProcess: failed to execvp:
xdg-settings
[1295:0615/165113.527416:ERROR:bus.cc(398)] Failed to connect to the bus: Failed to connect to socket /run/dbus/system_bus_socket: No such file or directory
[1295:0615/165113.527460:ERROR:bus.cc(398)] Failed to connect to the bus: Failed to connect to socket /run/dbus/system_bus_socket: No such file or directory
Main: restoring window state from [object Object]/window-state-plexamp-main.json
Main: Centering offscreen window.
Started Media Service: false
[1344:0615/165113.557720:ERROR:gpu_init.cc(481)] Passthrough is not supported, GL is egl, ANGLE is 
MESA-LOADER: failed to retrieve device information
MESA-LOADER: failed to open nvidia-drm: /usr/lib/x86_64-linux-gnu/GL/default/lib/dri/nvidia-drm_dri.so: cannot open shared object file: Permission denied (search paths /usr/lib/x86_64-linux-gnu/GL/default/lib/dri, suffix _dri)
failed to load driver: nvidia-drm
MESA-LOADER: failed to open kms_swrast: /usr/lib/x86_64-linux-gnu/GL/default/lib/dri/kms_swrast_dri.so: cannot open shared object file: Permission denied (search paths /usr/lib/x86_64-linux-gnu/GL/default/lib/dri, suffix _dri)
failed to load driver: kms_swrast
MESA-LOADER: failed to open swrast: /usr/lib/x86_64-linux-gnu/GL/default/lib/dri/swrast_dri.so: cannot open shared object file: Permission denied (search paths /usr/lib/x86_64-linux-gnu/GL/default/lib/dri, suffix _dri)
failed to load swrast driver
Linux: Registering MPRIS service.
[1353:0615/165113.599603:ERROR:command_buffer_proxy_impl.cc(126)] ContextResult::kTransientFailure: Failed to send GpuControl.CreateCommandBuffer.
APPIMAGE env is not defined, current application is not an AppImage
Couldn't install shortcut for MediaPlayPause.
Couldn't install shortcut for MediaPreviousTrack.
Couldn't install shortcut for MediaNextTrack.
Couldn't install shortcut for MediaStop.
```

</details>

**Comment by develop7** [comment]
Hey @simonxciv, I'm still affected by the issue. Couldn't help but notice the `--disable-gpu-sandbox` which seemed to crash it for me is still there in 4.8.0-r1

**Comment by user0753** [comment]
> Hey @simonxciv, the issue is still reproducing for me in 4.8.0-r1; could you reopen it? Couldn't help but notice the `--disable-gpu-sandbox` which seemed to crash it for me is still there

I had opened a thread on Plex forums yesterday about this here: https://forums.plex.tv/t/flatpak-version-of-plexamp-doesnt-open-on-wayland-w-nvidia/847807

I believe he originally said something about potentially breaking other NVIDIA configurations with that change. Unsure if he's speaking about driver versions or something else, like PRIME graphics configurations. In the meantime, he says disabling the Wayland socket in something like flatseal will make it fallback on xwayland and it works with that. 

I notice a discrepancy with his response to your pull request though. He said he'd disable Wayland by default in the pull request but ended up doing the opposite.

Side note, I discovered the music visualizer is laggy on native Wayland but not on xwayland. Also, same story with kwin window rules, I was able to force the title bar on with the xwayland window but not the native Wayland one.
